build/apply-patches

```diff
@@ -26,14 +24,49 @@ PATCH_DIR=${PATCH_DIR:-$(realpath "$BASE_DIR/patches/")}
 
 CUDF_DIR=${CUDF_DIR:-$(realpath "$BASE_DIR/thirdparty/cudf/")}
 
+# Apply pattches to CUDF is problematic in a number of ways. But ultimately it comes down to
+# making sure that a user can do development work in spark-rapids-jni without the patches
+# getting in the way
+# The operations I really want to support no matter what state CUDF is in are
+# 1) Build the repo from scratch
+# 2) Rebuild the repo without having to clean and start over
+# 3) upmerge to a new version of the plugin including updating the cudf submodule
+#
+# Building from scratch is simple. We want clean to unapply any patches and 
+# build to apply them. But if we want to rebuild without a clean we need to know what
+# state the CUDF repo is in. Did we apply patches to it or not. The fastest way to do this
+# is to save some state files about what happened. But a user could mess with CUDF directly
+# so we want to have ways to double check that they are indeed correct.
+
+FULLY_PATCHED_FILE="$CUDF_DIR/spark-rapids-jni.patch"
```

Comment:
there may technically be a patch that applies to non-cudf portion of spark-rapids, maybe should go to the top-level dir?

Reply:
Curious what's the use-case for patching outside of the submodule?  I thought we would checkin the change directly rather than a patch of that change.

Reply:
We only have a sole use case with the submodule patching right now. No doubt direct modifications of the source tree outside the module is preferrable. In terms of hypotheticals, there could be another submodule or some native dependency that  we want consume bypassing cudf which needs patching.

Reply:
Right now all patches apply only to CUDF. I don't want to try and make that more generic right now. If we do hit that situation I would rather have sub-directories under patches for each third party sub directory, and then apply/remove patches for each subdirectory separately.

Reply:
ok, separate subdirs work too